packages/react/src/runtimes/local/LocalThreadRuntime.tsx

```diff
@@ -235,7 +233,12 @@ export class LocalThreadRuntime implements ThreadRuntime {
     };
     this.repository.addOrUpdateMessage(parentId, message);
 
-    if (added && shouldContinue(message)) {
+    // Check if all tool calls have results
+    const allToolCallsHaveResults = message.content.every(
+      (c) => c.type !== "tool-call" || c.result,
```

Comment:
@Rajaniraiyn shouldContinue already checks for all tool calls having a result. this seems to be a duplicate check of the same thing

https://github.com/Yonom/assistant-ui/blob/c92547de23b878ec088f7c888dab4d1b3db1756e/packages/react/src/runtimes/local/shouldContinue.tsx#L6